app/uk/gov/hmrc/apiplatform/modules/submissions/domain/models/Submission.scala

```diff
@@ -97,23 +175,37 @@ object Submission {
       name: String
     ) extends Status
 
+    case class GrantedWithWarnings(
+      timestamp: DateTime,
+      name: String,
+      warnings: String
+    ) extends Status
+
     case class Submitted(
       timestamp: DateTime,
       requestedBy: String
     ) extends Status
 
+    case class Answering(
+      timestamp: DateTime,
+      completed: Boolean
+    ) extends Status
+
     case class Created(
       timestamp: DateTime,
       requestedBy: String
     ) extends Status
 
     def isLegalTransition(from: Submission.Status, to: Submission.Status): Boolean = (from, to) match {
-      case (c: Created,   s: Submitted) => true
-      case (s: Submitted, d: Declined)  => true
-      case (s: Submitted, g: Granted)   => true
-      case _                            => false
+      case (c: Created, a: Answering)               => true
+      case (Answering(_, true), s: Submitted)       => true
+      case (s: Submitted, d: Declined)              => true
+      case (s: Submitted, g: Granted)               => true
+      case (s: Submitted, w: GrantedWithWarnings)   => true
+      case (w: GrantedWithWarnings, d: Declined)    => true   // ? Maybe
```

Comment:
Did we decide if these 2 are valid?

Reply:
Not yet.  I've left them in as they cause no harm at the moment.  We need more input from design team as to how this warning state is meant to act.